src/AudioManager.ts

```diff
@@ -0,0 +1,158 @@
+  handleError?: (error: string) => void;
+}
+
+class AudioManager {
+  private audioInputsService: AudioInputsService;
+
+  private audioInputsBrowser: AudioInputsBrowser;
+
+  private audioOutputsService: AudioOutputsService;
+
+  private mediaAudio: MediaAudio;
+
+  private options: AudioManagerOptions;
+
+  constructor(options: AudioManagerOptions) {
+    this.options = {
+      duckVolumeLevel: options.duckVolumeLevel ?? 0,
+      normalVolumeLevel: options.normalVolumeLevel ?? 1,
+      sttService: options.sttService ?? "charisma/deepgram",
+      streamTimeslice: options.streamTimeslice ?? 100,
+      handleStartSTT: options.handleStartSTT,
+      handleStopSTT: options.handleStopSTT,
+      handleTranscript: options.handleTranscript,
+      handleError:
+        options.handleError ??
+        ((error: string) => console.error("Error:", error)),
+    };
+
+    this.audioInputsService = new AudioInputsService(
+      this.options.streamTimeslice,
+    );
+    this.audioInputsBrowser = new AudioInputsBrowser();
+    this.audioOutputsService = new AudioOutputsService();
+    this.mediaAudio = new MediaAudio();
+
+    // Listen to events from the AudioInputsService
+    this.audioInputsService.on("start", this.options.handleStartSTT);
+    this.audioInputsService.on("stop", this.options.handleStopSTT);
+    this.audioInputsService.on("transcript", this.options.handleTranscript);
+    this.audioInputsService.on("error", (error: string) =>
+      this.options.handleError?.(error),
+    );
+
+    // Listen to events from the AudioInputsBrowser
+    this.audioInputsBrowser.on("start", this.options.handleStartSTT);
+    this.audioInputsBrowser.on("stop", this.options.handleStopSTT);
+    this.audioInputsBrowser.on("transcript", this.options.handleTranscript);
+    this.audioInputsBrowser.on("error", (error: string) =>
+      this.options.handleError?.(error),
+    );
+  }
+
+  // **
+  // ** Audio Input ** //
+  // **
+  public startListening = (): void => {
+    if (this.options.sttService === "browser") {
+      this.audioInputsBrowser.startListening();
+    } else if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.startListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.options.duckVolumeLevel as number;
+    }
+  };
+
+  public stopListening = (): void => {
+    if (this.options.sttService === "browser") {
+      this.audioInputsBrowser.stopListening();
+    } else if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.stopListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.options.normalVolumeLevel as number;
+    }
+  };
+
+  public connect = (token: string): void => {
+    if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.connect(token);
+    }
+  };
+
+  public inputServiceResetTimeout = (timeout: number): void => {
```

Comment:
I don't think browserResetTimeout needs to be a separate function, can do it here?
Are audioInputsBrowser and audioInputsService isomorphic enough that we could set one this.inputsService in the constructor rather than having both?

Reply:
I can combine both resets to `resetTimeout` similar to startListening etc.
I'm not sure about combining them in the constructor because they have their own functions like browserIsSupported and connect.